### PR TITLE
feat(openchallenges): added unit tests to the image service

### PR DESCRIPTION
**Pull request by mdsage1**
## Description

Updated the unit test to cover api, model/dto and exception. Sonar was not working on my end so I'm unable to confirm the increase in coverage until the pr has been merged.

## Related Issue

Add unit test for getting a single organization [#1528](https://github.com/Sage-Bionetworks/sage-monorepo/issues/1528)

[Story] Write a unit test for the enpoint of the image service [#1456](https://github.com/Sage-Bionetworks/sage-monorepo/issues/1456)
feat: Create unit test for the image service [#1472](https://github.com/Sage-Bionetworks/sage-monorepo/pull/1472)
## Changelog

- Add

1. /workspaces/sage-monorepo/apps/openchallenges/image-service/src/test/java/org/sagebionetworks/exception/ImageHeightNotSpecifiedExceptionTest.java
2. /workspaces/sage-monorepo/apps/openchallenges/image-service/src/test/java/org/sagebionetworks/exception/SimpleChallengeGlobalExceptionTest.java

### CODE REVIEW

**Comment by mdsage1** [comment]
@tschaffter I think I've gotten this error before but now that I've deleted the node_modules folder I'm seeing this:

> BUILD FAILED in 13s
> 9 actionable tasks: 9 executed
> Failed to execute command: ./gradlew build 
> Error: Command failed: ./gradlew build 
>     at checkExecSyncError (node:child_process:885:11)
>     at execSync (node:child_process:957:15)
>     at runBuilderCommand (/workspaces/sage-monorepo/node_modules/@nxrocks/common/src/lib/core/jvm/utils.js:20:38)
>     at runBootPluginCommand (/workspaces/sage-monorepo/node_modules/@nxrocks/nx-spring-boot/src/utils/boot-utils.js:18:43)
>     at /workspaces/sage-monorepo/node_modules/@nxrocks/nx-spring-boot/src/executors/build/executor.js:10:54
>     at Generator.next (<anonymous>)
>     at /workspaces/sage-monorepo/node_modules/tslib/tslib.js:118:75
>     at new Promise (<anonymous>)
>     at Object.__awaiter (/workspaces/sage-monorepo/node_modules/tslib/tslib.js:114:16)
>     at buildExecutor (/workspaces/sage-monorepo/node_modules/@nxrocks/nx-spring-boot/src/executors/build/executor.js:8:20)
>     at /workspaces/sage-monorepo/node_modules/nx/src/command-line/run/run.js:81:23
>     at Generator.next (<anonymous>)
>     at fulfilled (/workspaces/sage-monorepo/node_modules/nx/node_modules/tslib/tslib.js:166:62)
>     at process.processTicksAndRejections (node:internal/process/task_queues:95:5)

Can you confirm that it's unrelated to node_modules and that I need to check my unit test code, please?

**Comment by tschaffter** [comment]
> I've deleted the node_modules folder

Did you recreate the folder with `workspace-install`? Even if it includes Node.js packages and you are working on a Java project, there are Node.js packages that are required for the monorepo to work, e.g. `nx`.

I believe that the error message is reported by a git hook but the error effectively comes from when `gradlew build`. Can you access the real error message thrown by gradle, e.g. by building the project that fail?

File: apps/openchallenges/image-service/src/test/java/org/sagebionetworks/openchallenges/image/service/api/ImageApiDelegateTest.java
Comment (tschaffter) [inline apps/openchallenges/image-service/src/test/java/org/sagebionetworks/openchallenges/image/service/api/ImageApiDelegateTest.java:1]: As a reminder, there is no need to add test files for code that has been automatically generated.

**Comment by mdsage1** [comment]
@tschaffter I added changes to unit tests for exception section of the project.

**Comment by tschaffter** [comment]
What is the coverage percentage for the two new files from the HTML report?

**Comment by mdsage1** [comment]
> What is the coverage percentage for the two new files from the HTML report?

@tschaffter I was unable to see any changes so I went ahead and creted the unit tests. I wrote it as a comment in the [Task](https://github.com/Sage-Bionetworks/sage-monorepo/issues/2062#issuecomment-1786277244) and forgot to write it here. I'm hoping to be able to see the changes in coverage once the unit tests are synced to the main branch.

**Comment by tschaffter** [comment]
@mdsage1 The coverage report is generated properly on my end. This PR brings the coverage of the image service to 30% (see screenshot).

## Coverage

### Global

image

### Exception package

image

**Pull request merged**